streamloader/src/main/scala/com/snowplowanalytics/snowplow/storage/bigquery/streamloader/StreamLoader.scala

```diff
@@ -50,23 +49,24 @@ object StreamLoader {
   private val MaxConcurrency = Runtime.getRuntime.availableProcessors * 16
   private val GroupByN       = 10
   private val TimeWindow     = 30.seconds
-  private val QueueSize      = 1024
 
   def run(e: Environment)(implicit CS: ContextShift[IO], C: Concurrent[IO], T: Timer[IO]): IO[Unit] =
     Resources.acquire(e).use { resources =>
       val eventStream = resources.source.evalMap(parse(resources.igluClient.resolver))
 
-      for {
-        queue <- Queue.bounded[IO, Set[ShreddedType]](QueueSize)
-        sinkBadGood = eventStream
-          .observeEither[StreamBadRow[IO], StreamLoaderRow[IO]](
-            resources.badRowsSink,
-            goodSink(resources, queue)
-          )
-          .void
-        sinkTypes = queue.dequeue.through(aggregateTypes(GroupByN, TimeWindow)).through(resources.typesSink)
-        _ <- sinkBadGood.merge(sinkTypes).compile.drain
-      } yield ()
+      val goodPipe: Pipe[IO, StreamLoaderRow[IO], Unit] =
+        _.observe(goodSink(resources))
+          .map(_.row.inventory)
+          .through(aggregateTypes(GroupByN, TimeWindow))
+          .through(resources.typesSink)
+
+      eventStream
+        .observeEither[StreamBadRow[IO], StreamLoaderRow[IO]](
+          resources.badRowsSink,
+          goodPipe
```

Comment:
It would be awesome if we can make it be just `observeEither(badSink, goodSink)`.